psl-core/src/main/java/org/linqs/psl/model/rule/AbstractRule.java

```diff
@@ -0,0 +1,34 @@
+/*
```

Comment:
Instead of making another class, I was just thinking of changing Rule from and interface to and abstract class.

Reply:
WeightedRule and UnweightedRule extend Rule.  The concrete classes are then a nice, orthogonal combination of arithmetic or logical, and unweighted or weighted.  To keep this, it seems we either leave Rule an interface, or change WeightedRule and UnweightedRule to abstract.  Thoughts?

Reply:
Actually, multiple inheritance is a no-go.  So, changing WeightedRule and UnweightedRule to abstract won't help.

Reply:
I think it makes most sense to break the dependency between Rule and WeightedRule/UnweightedRule.
(I also don't think Rule.isWeighted() should exist. One of the two purposes of the WeightedRule/UnweightedRule interfaces is so you can do `instanceof WeightedRule`. But that is just a rant, I'll probably be removing it after this patch comes in.)

With the dependency broken, we should be free to change Rule to an abstract class and store the name there.